### ISSUE TITLE
Prioritizing recall in mostly labeled dataset

### ISSUE THREAD

**Issue by readicculus** [AlexeyAB/darknet#7116]
I have a dataset of ~15,000 infrared hotspots/blobs that I can get up to about 90% recall for.  The reason I'm not just using a vision based blob detector is that the blobs aren't always blob shaped and sometimes there is a bit of noise that it seems yolo models do fine at ignoring.

I am mainly focused on recall because the dataset has some missing labels, I'd guess in all these images there are some 1,000-2,000 hotspots that are not labeled.  In general this data doesn't produce too many FP results so I am less concerned with increasing precision, what I really care about is that the model can find as many of the labeled hotspots as possible(and all hotspots when deployed).

So the idea I have is to modify the yolo loss function scaling parameters to put less weight on FP detections with high accuracy during training.  I'm not totally confident in my understanding of the yolo loss calculations so If someone can give me a bit of feedback or direction I would appreciate the help.

I'm less sure about how the anchor probability plays into the loss calculation, but is it correct to say that if I lower 𝜆noobj that anchors above a certain threshold in a noobj box will be penalized less?(essentially penalize FP during training less)

I think mosaicing may help too but our deployement system is not up to date for yolov4 meaning that I can't use a lot of stuff.  I think I should be fine with mosaicing though as this only happens during training?

Ideas on other ways to improve would be great,  realistically I should just go through and manually review "FP" detentions but I've been doing a lot of data work lately and I'm not trying to do any labeling work right now ;)

Examples of missing some labels in our ground truth data, this image is pretty obvious where hotspots are but not all are as clear(this is not detection, this is ground truth)
![test_kotz_2019_fl04_C_20190510_010032 647944_ir](https://user-images.githubusercontent.com/3818801/102032237-15b75c80-3d6d-11eb-98eb-7402669be562.png)
